plugins/UsersManager/tests/Integration/APITest.php

```diff
@@ -1174,6 +1174,29 @@ public function testSetUserAccessSetRoleAsArray()
         self::assertEquals([['site' => '1', 'access' => 'view']], $access);
     }
 
+    public function testSetUserAccessHandlesMultipleSitesWithMultiCurrentStateOfAccess()
+    {
+        $this->api->setUserAccess($this->login, [View::ID], [1]);
+        $this->api->setUserAccess($this->login, [View::ID], [1, 2]);
+
+        $access = $this->model->getSitesAccessFromUser($this->login);
+        self::assertEquals([['site' => '1', 'access' => 'view'], ['site' => '2', 'access' => 'view']], $access);
```

Comment:
To ensure roles are also changed and not only added, I think it's better to change the role in the test. Also adding a capability in the beginning will test that only providing a role, will remove all capabilities.
```suggestion
        $this->api->setUserAccess($this->login, [View::ID, TestCap1::ID], [1]);
        $this->api->setUserAccess($this->login, [Admin::ID], [1, 2]);

        $access = $this->model->getSitesAccessFromUser($this->login);
        self::assertEquals([['site' => '1', 'access' => 'admin'], ['site' => '2', 'access' => 'admin']], $access);
```